### ISSUE TITLE
some thoughts about API Platform:  custom route,collecting data use Symfony form instead of serializer

### ISSUE THREAD

**Issue by videni** [api-platform/api-platform#648]
I am new to API platform, just use it for a week.

1. handle multy/form-data
currently I disable  auto-generated POST route, replace it with my own implementation, I have to deal with content negotiation,  error handling(format error message) my self, because api platform event listeners are only available for auto-generated routes.

2.  Can't add swagger description to my action. I have to write them manually , then merge it with the generated one. this is not nice.

3. create a entity that need to select a relation， for example， when create a product,  one needs to select a  taxon  which the product belong.  product -> taxon  is a many-to-many relation. What I do is to create a Symfony form do handle this, write a custom route outside of API platform flow. 

In summary , using serializer to collection data from request to entity  is not good enough , I think  it would be great if API platform can : 

1.  Let us configure the way we collection data,  serializer or Symfony form. Symfony form can provide more advance features.  

2. Replace a route with custom implementation,   [creating-custom-operations-and-controllers](https://api-platform.com/docs/core/operations#creating-custom-operations-and-controllers)  doesn't fulfill my need, for no request object passed to __invoke function,  I may need to check such as whether a taxon is existed in database manually.

**Comment by soyuka** [comment]
> 

Replace a route with custom implementation, creating-custom-operations-and-controllers doesn't fulfill my need, for no request object passed to __invoke function, I may need to check such as whether a taxon is existed in database manually.

You can totally do this

**Comment by gries** [comment]
Make use of the event System for that.
If you want to handle html-forms don‘t eitler use plain symfony or submit your Firma using JS and send json data.

**Comment by videni** [comment]
@soyuka , thank you so much, I know how to do that now.
